### PR TITLE
Updated URLs to be Markdown links.

### PR DESCRIPTION
**Pull request by twostraws**
This changes the links to be clickable.

### Motivation:

It reads better, is easier to use, and is less annoying for VoiceOver to read.

### Modifications:

I made the URLs into Markdown links.

### Result:

The URLs are now clickable.

### CODE REVIEW

**Review by daveverwer — APPROVED** [review]
💯

File: _posts/2022-05-19-mentorship-2022.md [hunk 1]
```diff
@@ -38,11 +38,11 @@ Though Amritpan formed an effective strategy for breaking down Swift evolution p
...
+[You can read Amritpan’s full reflection on her mentorship journey here](https://forums.swift.org/t/swift-mentorship-compiler-language-design/52522).
 
 
 ## Changes to the mentorship program
 
 The Diversity in Swift work group is excited to continue offering the Swift Mentorship Program to welcome and support more programmers in the Swift community. After the 12-week cohort over the summer, we’ll continue to offer mentorship on a rolling bases for starter bug contributions, so new mentors and new mentees can submit applications to join throughout the year.
 
-You can find out more about the Swift mentorship program, and instructions for how to apply at [Swift.org/mentorship](/mentorship/).
\ No newline at end of file
+You can find out more about the Swift mentorship program, and instructions for how to apply at [swift.org/mentorship](/mentorship/).
```
Comment (hborla) [inline _posts/2022-05-19-mentorship-2022.md:48]: This was intentional; Swift.org is capitalized elsewhere on the website (e.g. on Swift.org/mentorship/ and Swift.org/website/)

Comment (twostraws) [reply _posts/2022-05-19-mentorship-2022.md:48]: Ah! Sorry about that. I was following other examples I had seen on https://www.swift.org/download and https://www.swift.org/blog/swift-docc. I'll revert that change, and we can look at adjusting the others.

**Review by shahmishal — APPROVED** [review]

**Comment by shahmishal** [comment]
@swift-ci test

**Pull request merged**